kustomize/commands/edit/fix/splitpatches.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package fix
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+// splitPatches splits patches that contain multiple documents into multiple files and updates the kustomization file's
+// Patch field to point to the new files.
+func splitPatches(fSys filesys.FileSystem, k *types.Kustomization, w io.Writer) error {
+	yamlSeparatorRegexp := regexp.MustCompile(`(?:^|\n)---.*`) // Matches any line that starts with ---
```

Comment:
Maybe I should update that other regex and expose it.

The other one is `\n---.*\n` which doesn't account for a file that begins with `---` or ends with `---`.  Whoever wrote that one clearly doesn't Regex very well (spoiler, it was me :sweat_smile:)

For example, it would split this...
```
---
something
---
something else 
---
```

into this:
```
---
something
```
and this:
```
something else 
---
```

(Notice the remaining `---` separators)

I believe `(?:^|\n)---.*` is better because it will match `---` at any new line, including the first and last line of the file.

The split result with the new regex would be:
```
something
```
and
```
something else 
```

Reply:
Ok, that makes sense. I'm in favour of updating the other one, since that instance is more important than the "fix" code! I remembered that strictly speaking there are document footers in yaml too, and I wanted to refresh my memory on what the whole flow can be, so I went to the spec: https://yaml.org/spec/1.2.2/#91-documents. I don't think we should start handling footers, but strictly speaking your example has three documents: the two the new regex will split out, plus an empty explicit document at the end. Maybe the helper we expose should preserve the third as well, or have a 'drop empty' option?